core/test/database/sqlcipher_tests.cpp

```diff
@@ -104,12 +95,43 @@ TEST(SQLCipherTest, SanityCheck) {
         EXPECT_NO_THROW((session << "SELECT id, name FROM test_table", soci::into(value_0), soci::into(value_1)));
 
         session.close();
+    }
+}
 
-        session.open(*factory, parameters);
+// Preventing terminating in environments that don't properly define __cplusplus macros
+// TODO: remove this check after we migrate to VS2017 and provide /Zc:__cplusplus option 
+// https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
+#if __cplusplus >= 201103L
+TEST(SQLCipherTest, ThrowIfWrongPassword) {
```

Comment:
Hm, I don’t know how to feel about that. Tests shouldn’t be compiler-dependent like that. Also, what is that CPP conditional expression for? I don’t get the part with VS2017.

Reply:
soci_core throws exceptions on the destructors. Starting from C++11 throwing exception in destructor should terminate program, unless the destructor was marked as `noexcept(false)` this is what soci_core is doing:
```
#if __cplusplus >= 201103L
#define SOCI_ONCE_TEMP_TYPE_NOEXCEPT noexcept(false)
#else
#define SOCI_ONCE_TEMP_TYPE_NOEXCEPT
#endif

and latter 

class SOCI_DECL once_temp_type
{
public:
...    
    ~once_temp_type() SOCI_ONCE_TEMP_TYPE_NOEXCEPT;
....
```
According to the article, all Visual Studios up to VS2017 15.7 defining __cplusplus as 199711L 
In other words Live Desktop on Windows will terminate if someone will try open encrypted DB with wrong password, don't know what to do here... either use another db driver, or migrate completely to VS2017 ...tuft choose